### PR TITLE
Sscs 9439 action postponement request event   grant postponement

### PR DESCRIPTION
**Pull request by gmikerlemos**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-9439

### Change description ###

Action Postponement Request Event - Grant Postponement

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java [hunk 1]
```diff
@@ -39,32 +50,107 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
...
+        sscsCaseData.setInterlocReviewState(InterlocReviewState.REVIEW_BY_JUDGE.getId());
+        sscsCaseData.setInterlocReferralReason(InterlocReferralReason.REVIEW_POSTPONEMENT_REQUEST.getId());
+        sscsCaseData.setPostponementRequest(PostponementRequest.builder().unprocessedPostponementRequest(YesNo.YES)
+                .build());
+    }
+
+    private void addDirectionNotice(SscsCaseData caseData) {
+        DocumentLink url = caseData.getPreviewDocument();
+        SscsDocumentTranslationStatus documentTranslationStatus = caseData.isLanguagePreferenceWelsh() ? SscsDocumentTranslationStatus.TRANSLATION_REQUIRED : null;
+        footerService.createFooterAndAddDocToCase(url, caseData, AUDIO_VIDEO_EVIDENCE_DIRECTION_NOTICE,
```
Comment (rasikasri) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java:125]: This is wrong event type. Has to be ACTION_POSTPONEMENT_REQUEST

Comment (gmikerlemos) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java:125]: Well spotted. We already have POSTPONEMENT_REQUEST though so I am creating a new type for the directions notice. Nice one :)

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java [hunk 2]
```diff
@@ -39,32 +50,107 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
...
+    private boolean isGrantPostponement(PostponementRequest postponementRequest) {
+        return postponementRequest.getActionPostponementRequestSelected().equals(GRANT.getValue());
+    }
+
+    private boolean isSendToJudge(PostponementRequest postponementRequest) {
+        return postponementRequest.getActionPostponementRequestSelected().equals(SEND_TO_JUDGE.getValue());
+    }
+
+    private static LocalDate getLocalDate(String dateStr) {
+        return LocalDate.parse(dateStr, DateTimeFormatter.ofPattern("dd-MM-yyyy"));
```
Comment (rasikasri) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java:154]: Is this correct date formatter? I get a parsing error for this format. I think ccd has yyyy-MM-dd format. Also better to move this to a constant as this is used in three different places in this class.

Comment (gmikerlemos) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java:154]: well spotted the wrong doc type. I had to create a new one as Postponement Request is the doc type for the doc in Action FE

Comment (gmikerlemos) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java:154]: this is what I could find wide spread in tribunals

Comment (gmikerlemos) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java:154]: Process A/V was my template

Comment (rasikasri) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java]: I think dd-MM-yyyy format is used to write date values, but when you read date string coming from ccd, you might need to use yyyy-MM-dd. Please have a look at RequestHearingRecordingAboutToSubmitHandler, which is reading same date fields that you are reading here.

Comment (gmikerlemos) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java]: fixed

**Review by rasikasri — APPROVED** [review]

**Pull request merged**